From c9842d334e166a3b04c81dd62b4e7b2317c8421b Mon Sep 17 00:00:00 2001
From: "Manuel A. Fernandez Montecelo" <manuel.montezelo@gmail.com>
Date: Mon, 28 Mar 2016 23:20:30 +0100
Subject: [PATCH 1/2] Provide bits/fcntl.h for RISC-V

This file provides an implementation similar to other 64-bit architectures, so
F_GETLK64, F_SETLK64 and F_SETLKW64 are the same as the non-64 counterparts and
__O_LARGEFILE is defined correctly.

(cherry picked from commit 0f3c27e8f08b0ba5d1640e6ffef56213b7c5eb31)
---
 glibc/sysdeps/unix/sysv/linux/riscv/bits/fcntl.h | 36 ++++++++++++++++++++++++
 1 file changed, 36 insertions(+)
 create mode 100644 glibc/sysdeps/unix/sysv/linux/riscv/bits/fcntl.h

diff --git a/glibc/sysdeps/unix/sysv/linux/riscv/bits/fcntl.h b/glibc/sysdeps/unix/sysv/linux/riscv/bits/fcntl.h
new file mode 100644
index 0000000..6a00f26
--- /dev/null
+++ b/glibc/sysdeps/unix/sysv/linux/riscv/bits/fcntl.h
@@ -0,0 +1,36 @@
+/* O_*, F_*, FD_* bit values for Linux / RISC-V.
+   Copyright (C) 2016 Free Software Foundation, Inc.
+   This file is part of the GNU C Library.
+
+   The GNU C Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+
+   The GNU C Library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with the GNU C Library; if not, see
+   <http://www.gnu.org/licenses/>.  */
+
+#ifndef	_FCNTL_H
+# error "Never use <bits/fcntl.h> directly; include <fcntl.h> instead."
+#endif
+
+#include <bits/wordsize.h>
+
+/* In 64-bit ISA files are always with 64bit off_t and F_*LK64 are the same as
+ * non-64-bit versions.  It will need to be revised for 128-bit.  */
+#if __WORDSIZE == 64
+# define __O_LARGEFILE	0
+
+# define F_GETLK64	5	/* Get record locking info.  */
+# define F_SETLK64	6	/* Set record locking info (non-blocking).  */
+# define F_SETLKW64	7	/* Set record locking info (blocking).	*/
+#endif
+
+/* Include generic Linux declarations.  */
+#include <bits/fcntl-linux.h>
-- 
2.7.4

